src/gui/editors/PianoRoll.cpp

```diff
@@ -3872,6 +3908,8 @@ void PianoRoll::updateSemiToneMarkerMenu()
 
 	emit semiToneMarkerMenuScaleSetEnabled( ! scale.isEmpty() );
 	emit semiToneMarkerMenuChordSetEnabled( ! chord.isEmpty() );
+
+	PianoKeyboard::setScaleType( m_scaleModel.currentText() );
```

Comment:
Ouch! `currentText()` is not reliable because translations will break it. Use `value()` instead.

@BaraMGB:
> I selected scale: DUR. But all keys play note C. In different Rows different Cs (C3,C4,C5) but only C. If I choose an other note only this note will be played. Perhaps I have done something wrong?